core/code/android.js

```diff
@@ -112,6 +112,18 @@ window.runOnAndroidBeforeBoot = function () {
       android.saveFile(filename || '', dataType || '*/*', data);
     };
   }
+
+  if (android.intentPosLink) {
+    window.renderPortalUrl = function (lat, lng, title, guid) {
+      // one share link option - and the android app provides an interface to share the URL,
+      // share as a geo: intent (navigation via google maps), etc
+
+      var shareLink = $('<a>').text('Share portal').click(function () {
+        android.intentPosLink(lat, lng, window.map.getZoom(), title, true);
+      });
+      $('linkdetails').append($('<aside>').append(shareLink));
+    };
+  }
```

Comment:
this pattern (creating a HTMLElement, and fill if afterward) looks wrong to me in this context.
What about hooking on `renderDetailUpdated` instead ? Or make the function return a `<div>` directly ?

Reply:
> What about hooking on renderDetailUpdated instead?

Thwt would be perfect, but there is a condition: https://github.com/IITC-CE/ingress-intel-total-conversion/blob/4d46ab7178bc2c07252c05758a393ac34b037b2d/core/code/portal_detail_display.js#L155-L157

> Or make the function return a `<div>` directly ?

Well, there are 2 elements.. Put them under one div?

Reply:
> Thwt would be perfect, but there is a condition: 

yes, the hook doesn't match so that won't work

> Well, there are 2 elements.. Put them under one div? 

there is only one `<div class="linkDetails"/>` so that matches

note: Raw portal also inject it's "Raw data" button in this div

(anyway, that's only a pattern I don't commonly use. I prefer function that returns element smth that function that display smth when the choice)